INTRODUCTION.md

```diff
@@ -1258,70 +1376,234 @@ The purged messages in queue will fail with error code set to
 
 
 
+## Compatibility
 
+### Broker version compatibility
 
-## Appendix
-
-### Test details
-
-#### Test1: Produce to two brokers, two partitions, required.acks=2, 100 byte messages
-
-Each broker is leader for one of the two partitions.
-The random partitioner is used (default) and each broker and partition is
-assigned approximately 250000 messages each.
-
-**Command:**
-
-    # examples/rdkafka_performance -P -t test2 -s 100 -c 500000 -m "_____________Test1:TwoBrokers:500kmsgs:100bytes" -S 1 -a 2
-	....
-    % 500000 messages and 50000000 bytes sent in 587ms: 851531 msgs/s and 85.15 Mb/s, 0 messages failed, no compression
-
-**Result:**
-
-Message transfer rate is approximately **850000 messages per second**,
-**85 megabytes per second**.
-
-
-
-#### Test2: Produce to one broker, one partition, required.acks=0, 100 byte messages
+librdkafka supports all released Apache Kafka broker versions since 0.8.0.0.0,
+but not all features may be available on all broker versions since some
+features rely on newer broker functionality.
 
-**Command:**
+**Current defaults:**
+ * `api.version.request=true`
+ * `broker.version.fallback=0.9.0.0`
+ * `api.version.fallback.ms=0` (never revert to `broker.version.fallback`)
 
-    # examples/rdkafka_performance -P -t test2 -s 100 -c 500000 -m "_____________Test2:OneBrokers:500kmsgs:100bytes" -S 1 -a 0 -p 1
-	....
-	% 500000 messages and 50000000 bytes sent in 698ms: 715994 msgs/s and 71.60 Mb/s, 0 messages failed, no compression
+Depending on what broker version you are using, please configure your
+librdkafka based client as follows:
 
-**Result:**
+#### Broker version >= 0.10.0.0 (or trunk)
 
-Message transfer rate is approximately **710000 messages per second**,
-**71 megabytes per second**.
-
-
-
-#### Test3: Produce to two brokers, two partitions, required.acks=2, 100 byte messages, snappy compression
-
-**Command:**
-
-	# examples/rdkafka_performance -P -t test2 -s 100 -c 500000 -m "_____________Test3:TwoBrokers:500kmsgs:100bytes:snappy" -S 1 -a 2 -z snappy
-	....
-	% 500000 messages and 50000000 bytes sent in 1672ms: 298915 msgs/s and 29.89 Mb/s, 0 messages failed, snappy compression
-
-**Result:**
-
-Message transfer rate is approximately **300000 messages per second**,
-**30 megabytes per second**.
+For librdkafka >= v1.0.0 there is no need to set any api.version-related
+configuration parameters, the defaults are tailored for broker version 0.10.0.0
+or later.
 
+For librdkafka < v1.0.0, please specify:
+```
+api.version.request=true
+api.version.fallback.ms=0
+```
```

Comment:
in the client examples, we still specify:

```
    'broker.version.fallback': '0.10.0.0',
    'api.version.fallback.ms': 0,
```

by my reading of these docs, that is not required. is that correct?

if so, we should update our examples. I also think it'd be worth updating the cut-n-paste docs in confluent cloud to have these commented out with an explanation of when they're needed (pre librdkafka 1.0 only) as they come across as a bit weird.

Reply:
Our examples should definitely be updated, as for CCloud, that's trickier: not sure multiple variants is less confusing, and it is likely that the user does not know the version of its Kafka client.